mcs/class/corlib/Makefile

```diff
@@ -43,12 +43,12 @@ ifndef MOBILE_STATIC
 REFERENCE_SOURCES_FLAGS += -d:FEATURE_REMOTING,MONO_COM,FEATURE_COMINTEROP,FEATURE_ROLE_BASED_SECURITY
 endif
 
-LOCAL_MCS_FLAGS = -unsafe -nostdlib -nowarn:612,618 -d:INSIDE_CORLIB,MONO_CULTURE_DATA -d:LIBC $(REFERENCE_SOURCES_FLAGS)
+LOCAL_MCS_FLAGS = -unsafe -nostdlib -nowarn:612,618 -d:INSIDE_CORLIB,MONO_CULTURE_DATA,NET_4_6 -d:LIBC $(REFERENCE_SOURCES_FLAGS)
```

Comment:
This isn't the correct way of setting the define, you need to create a new profile similar to [net_4_5.make](https://github.com/mono/mono/blob/master/mcs/build/profiles/net_4_5.make) and plumb it throughout `configure.ac` etc.